File: clients/android/app/src/main/java/network/buildit/modules/federation/FederationModule.kt
package network.buildit.modules.federation

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Public
import network.buildit.core.modules.BuildItModule
import network.buildit.core.modules.ModuleRoute
import network.buildit.core.nostr.NostrEvent
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Federation module — manages AP and AT Protocol bridge settings.
 * Federation is handled server-side by the federation worker;
 * this module provides the settings UI and status display.
 */
@Singleton
class FederationModule @Inject constructor() : BuildItModule {
    override val identifier = "federation"
    override val version = "1.0.0"
    override val dependencies = emptyList<String>()
    override val displayName = "Federation"
    override val description = "Bridge public posts to Mastodon and Bluesky"

    override suspend fun initialize() {
        // Load cached federation status
    }

    override suspend fun handleEvent(event: NostrEvent): Boolean {
        // Federation is worker-side — no local event handling
        return false
    }

    override fun getNavigationRoutes(): List<ModuleRoute> {
        return listOf(
            ModuleRoute(
                route = "federation-settings",
                title = "Federation",
                icon = Icons.Default.Public,
            ) { FederationSettingsScreen() }
        )
    }

    override fun getHandledEventKinds(): List<Int> = emptyList()
}

File: clients/ios/BuildIt/Modules/Federation/FederationModule.swift
// FederationModule.swift
// BuildIt - Federation bridge for ActivityPub and AT Protocol
//
// Manages federation settings and status for cross-posting
// public content to Mastodon and Bluesky.

import SwiftUI

/// Federation module — manages AP and AT Protocol bridge settings
@MainActor
public final class FederationModule: BuildItModule {
    public static let identifier = "federation"
    public static let version = "1.0.0"
    public static let dependencies: [String] = []

    public init() throws {}

    public func initialize() async throws {
        // Load cached federation status
    }

    public func handleEvent(_ event: NostrEvent) async {
        // Federation is worker-side — no local event handling needed
    }

    public func getViews() -> [ModuleView] {
        return [
            ModuleView(
                id: "federation-settings",
                title: "Federation",
                icon: "globe",
                destination: AnyView(FederationSettingsView())
            )
        ]
    }

    public func cleanup() async {}

    public func isEnabled(for groupId: String?) -> Bool {
        // Federation is user-level, always available
        return true
    }

    public func enable(for groupId: String?) async throws {}
    public func disable(for groupId: String?) async {}
}
